File: proyectos_c#/juegos/JuegoSnake/JuegoSnake/Cola.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace JuegoSnake
{
    public class Cola : Objeto
    {
        private Cola siguiente;
        public Cola(int tamano, int x, int y):base(tamano, x, y)
        {
            this.siguiente = null;
        }

        public void setxy(int x, int y)
        {
            if (this.siguiente != null)
            {
                this.siguiente.setxy(this.x, this.y);
            }
            this.x = x;
            this.y = y;
        }
        public int verX()
        {
            return this.x;
        }
        public int verY()
        {
            return this.y;
        }
        public void meter()
        {
            if (this.siguiente == null)
            {
                this.siguiente = new Cola(this.tamano, this.x, this.y);
            }
            else
            {
                this.siguiente.meter();
            }
        }

        public Boolean colisiona(Objeto obj)
        {
            if (this.interseccion(obj))
            {
                return true;
            }
            else if (this.siguiente != null)
            {
                return this.siguiente.colisiona(obj);
            }
            else
            {
                return false;
            }
        }

        public Boolean revision()
        {
            if (this.siguiente != null)
            {
                return this.siguiente.colisiona(this);
            }
            else
            {
                return false;
            }
        }
        public void dibujar(Graphics g)
        {
            if (this.siguiente != null)
            {
                this.siguiente.dibujar(g);
            }
            g.FillRectangle(new SolidBrush(Color.Blue), this.x, this.y, this.tamano, this.tamano);
        }
    }
}

File: proyectos_basic/juegos/SnakeBasic/SnakeBasic/Cola.vb
Public Class Cola
    Inherits Objeto
    Dim siguiente As Cola
    Public Sub New(tamano As Integer, x As Integer, y As Integer)
        MyBase.New(tamano, x, y)
        Me.siguiente = Nothing
    End Sub
    Public Sub setxy(x As Integer, y As Integer)
        If (Me.siguiente IsNot Nothing) Then
            Me.siguiente.setxy(Me.x, Me.y)
        End If
        Me.x = x
        Me.y = y
    End Sub
    Public Function verX()
        Return Me.x
    End Function
    Public Function verY()
        Return Me.y
    End Function
    Public Sub meter()
        If (Me.siguiente Is Nothing) Then
            Me.siguiente = New Cola(Me.tamano, Me.x, Me.y)
        Else
            Me.siguiente.meter()
        End If
    End Sub
    Public Function colisiona(obj As Objeto)
        If (Me.interseccion(obj)) Then
            Return True
        ElseIf (Me.siguiente IsNot Nothing) Then
            Return Me.siguiente.colisiona(obj)
        Else
            Return False
        End If
    End Function
    Public Function revision()
        If (Me.siguiente IsNot Nothing) Then
            Return Me.siguiente.colisiona(Me)
        Else
            Return False
        End If
    End Function
    Public Sub dibujar(g As Graphics)
        If (Me.siguiente IsNot Nothing) Then
            Me.siguiente.dibujar(g)
        End If
        g.FillRectangle(New SolidBrush(Color.Blue), Me.x, Me.y, Me.tamano, Me.tamano)
    End Sub
End Class


